bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/servlet/AccountVerificationServlet.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.blink.internal.servlet;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.blink.internal.dto.BlinkAccount;
+import org.openhab.binding.blink.internal.handler.AccountHandler;
+import org.openhab.binding.blink.internal.service.AccountService;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AccountVerificationServlet} class provides the servlet for validation of the 2-factor authentication pin
+ * provided by blink.
+ *
+ * The content is loaded from the componentbundle resources (see validation.html)
+ *
+ * @author Matthias Oesterheld - Initial contribution
+ */
+@NonNullByDefault
+public class AccountVerificationServlet extends HttpServlet {
+
+    public static final long serialVersionUID = 666L;
+
+    private static final String SETTINGS_THINGS = "/settings/things/";
+    private final Logger logger = LoggerFactory.getLogger(AccountVerificationServlet.class);
+    private final HttpService httpService;
+    private final AccountHandler accountHandler;
+    private final AccountService blinkService;
+    private final String servletUrl;
+    private final BundleContext bundleContext;
+
+    public AccountVerificationServlet(HttpService httpService, BundleContext bundleContext,
+            AccountHandler accountHandler, AccountService blinkService) {
+        this.httpService = httpService;
+        this.bundleContext = bundleContext;
+        this.accountHandler = accountHandler;
+        this.blinkService = blinkService;
+
+        try {
+            servletUrl = "/blink/"
+                    + URLEncoder.encode(accountHandler.getThing().getUID().getId(), StandardCharsets.UTF_8);
+            httpService.registerServlet(servletUrl, this, null, httpService.createDefaultHttpContext());
```

Comment:
Make the servlet name unique to prevent issues when multiple accounts are used.

See the [community topic](https://community.openhab.org/t/blink-binding/132328/47)

```suggestion
            servletUrl = "/blink/"
                    + URLEncoder.encode(accountHandler.getThing().getUID().getId(), StandardCharsets.UTF_8);

            Hashtable<Object, Object> initParams = new Hashtable<>();
            initParams.put("servlet-name", servletUrl);

            httpService.registerServlet(servletUrl, this, initParams, httpService.createDefaultHttpContext());
```